### PR TITLE
Sql rename col for aggregation columns and round function

### PR DESCRIPTION
**Pull request by Macbeth98**
# Description
- Renaming of columns with alias `As` keyword implemented for aggregation columns and round function.
- Implemented the loop for renaming of cols at `colReq.RenameColumns != null` in `segaggs.go`.

Feature **#Adds Aliasing**

# Testing
- Added a test case in `sqlparser_test.go` for testing of renaming of round function. For aggregation, the tests are already there.

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

**Comment by AndrewHess** [comment]
Is this the same as #248 plus the ability to rename the output field? If so it might be better to wait until that PR is merged, then either merge in those changes or rebase on top of them, so that it's easier to see what this PR is actually adding

**Comment by Macbeth98** [comment]
@AndrewHess This one is built on top of my previous PR. So it is the same as #248, plus the ability to rename the output field. I can push these changes also to that branch and close this PR if needed.

**Comment by AndrewHess** [comment]
Yep feel free to do either one of these:
1. Incorporate all the changes into that PR and close this PR

OR

2. Mark this as a draft PR and once the other PR is merged, either merge those changes into this PR or rebase this PR on top of those changes. Then mark this PR as ready for review

**Comment by Macbeth98** [comment]
@AndrewHess I think I will do the second option as that PR is almost ready to be merged. So I will make this as a draft PR and once the other one is merged, I will merge those changes and then mark this PR as ready for review.

**Comment by Macbeth98** [comment]
@AndrewHess Thank you very much for reviewing and merging the previous PR #248.
- Now, this PR is ready for review.
- This PR contains the renaming or aliasing for aggregation columns like sum, count, and average, as well as for round functions.

Please review it and let me know of any further changes. Thank you!

**Comment by Macbeth98** [comment]
@AndrewHess Can you please review this whenever you can? Thank you very much!🙏

File: pkg/segment/aggregations/segaggs.go
Comment (AndrewHess) [inline pkg/segment/aggregations/segaggs.go:172]: nit: delete this instead of commenting it out

**Review by AndrewHess — COMMENTED** [review]
This looks good. It's ready to merge once my one comment is addressed

File: pkg/segment/aggregations/segaggs.go
Comment (Macbeth98) [reply pkg/segment/aggregations/segaggs.go:172]: Sure, got it!

**Comment by Macbeth98** [comment]
@AndrewHess No Worries! Thank you very much for taking the time to review. 

I have removed the commented code as you have suggested. The PR is ready to be merged I believe.

Let me know of any further changes.

**Review by AndrewHess — APPROVED** [review]

**Pull request merged**
